pqAbstractItemViewEventTranslatorBase.cxx

```diff
@@ -0,0 +1,291 @@
+/*=========================================================================
+
+   Program: ParaView
+   Module:    pqAbstractItemViewEventTranslatorBase.cxx
+
+   Copyright (c) 2005,2006 Sandia Corporation, Kitware Inc.
+   All rights reserved.
+
+   ParaView is a free software; you can redistribute it and/or modify it
+   under the terms of the ParaView license version 1.2.
+
+   See License_v1.2.txt for the full ParaView license.
+   A copy of this license can be obtained by contacting
+   Kitware Inc.
+   28 Corporate Drive
+   Clifton Park, NY 12065
+   USA
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR
+CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+========================================================================*/
+#include "pqAbstractItemViewEventTranslatorBase.h"
+#include "pqEventTypes.h"
+
+#include <QEvent>
+#include <QKeyEvent>
+#include <QAbstractItemView>
+#include <QVariant>
+
+//-----------------------------------------------------------------------------
+pqAbstractItemViewEventTranslatorBase::pqAbstractItemViewEventTranslatorBase(QObject* parentObject)
+  : Superclass(parentObject)
+{
+  this->Editing = false;
+  this->AbstractItemView = NULL;
+  this->ModelItemCheck = NULL;
+  this->AbstractItemViewMouseTracking = false;
+  this->Checking = false;
+}
+
+//-----------------------------------------------------------------------------
+pqAbstractItemViewEventTranslatorBase::~pqAbstractItemViewEventTranslatorBase()
+{
+  if (this->AbstractItemView != NULL)
+    {
+    this->AbstractItemView->setMouseTracking(this->AbstractItemViewMouseTracking);
+    }
+}
+
+//-----------------------------------------------------------------------------
+bool pqAbstractItemViewEventTranslatorBase::translateEvent(
+  QObject* object, QEvent* event, int eventType, bool& error)
+{
+  // Recover corrected abstract item view
+  QAbstractItemView* abstractItemView = NULL;
+  if (!this->findCorrectedAbstractItemView(object, abstractItemView) || !abstractItemView)
+    {
+    return false;
+    }
+
+  // Don't try to record events for QComboBox implementation details
+  if(QString(abstractItemView->metaObject()->className()) == "QComboBoxListView")
```

Comment:
nice one, did not know inherits. ( fyi, there is a lot of usage of this in QtTesting, should i change it ? )